### PR TITLE
add more details when error on call_with_conn

### PR DESCRIPTION
**Pull request by hi-rustin**
close https://github.com/diesel-rs/diesel/issues/2646

### CODE REVIEW

**Comment by hi-rustin** [comment]
Look like:
```sh
➜  doesel-test ../diesel/target/debug/diesel migration run --database-url postgres://localhost/lemmy
Could not connect to database via `postgres://localhost/lemmy`: fe_sendauth: no password supplied

➜  doesel-test 

```

File: diesel_cli/src/database.rs [hunk 1]
```diff
@@ -134,7 +134,7 @@ macro_rules! call_with_conn {
         $($func:ident)::+ ($($args:expr),*)
     ) => {
         match crate::database::InferConnection::establish(&$database_url)
-            .unwrap_or_else(handle_error)
+            .unwrap_or_else(|err| {crate::database::handle_error_with_database_url(&$database_url,err)})
```
Comment (weiznich) [inline diesel_cli/src/database.rs:137]:
```suggestion
            .unwrap_or_else(|err| {crate::database::handle_error_with_database_url(&$database_url, err)})
```

**Review by weiznich — COMMENTED** [review]
I would like to see a test case for this.

**Comment by hi-rustin** [comment]
@weiznich  Do you have any suggestions about run cli tests locally?

I didn't find it in the contribution guide.

**Comment by hi-rustin** [comment]
> @weiznich Do you have any suggestions about run cli tests locally?
> 
> I didn't find it in the contribution guide.

Got it from GitHub actions.

**Comment by weiznich** [comment]
Perfect :+1:, thanks

**Pull request merged**